Account/FxAPushMessageHandler.swift

```diff
@@ -89,65 +87,44 @@ extension FxAPushMessageHandler {
                     }
-                    var messages: [PushMessage] = []
-
-                    // It's possible one of the messages is a device disconnection
-                    // in that case, we have an async call to get the name of the device
-                    // we should make sure not to resolve our own value before that name retrieval
-                    // is done
-                    var waitForClient: Deferred<Maybe<String>>?
-                    for event in events {
-                        switch event {
-                        case .commandReceived(let deviceCommand):
-                            switch deviceCommand {
-                            case .tabReceived(_, let tabData):
-                                let title = tabData.entries.last?.title ?? ""
-                                let url = tabData.entries.last?.url ?? ""
-                                messages.append(PushMessage.commandReceived(tab: ["title": title, "url": url]))
-                                if let json = try? accountManager.gatherTelemetry() {
-                                    let events = FxATelemetry.parseTelemetry(fromJSONString: json)
-                                    events.forEach { $0.record(intoPrefs: self.profile.prefs) }
-                                }
-                            }
-                        case .deviceConnected(let deviceName):
-                            messages.append(PushMessage.deviceConnected(deviceName))
-                        case let .deviceDisconnected(deviceId, isLocalDevice):
-                            if isLocalDevice {
-                                // We can't disconnect the device from the account until we have access to the application, so we'll handle this properly in the AppDelegate (as this code in an extension),
-                                // by calling the FxALoginHelper.applicationDidDisonnect(application).
-                                self.profile.prefs.setBool(true, forKey: PendingAccountDisconnectedKey)
-                                messages.append(PushMessage.thisDeviceDisconnected)
-                            }
 
-                            guard let profile = self.profile as? BrowserProfile else {
-                                // We can't look up a name in testing, so this is the same as not knowing about it.
-                                messages.append(PushMessage.deviceDisconnected(nil))
-                                break
-                            }
+                    switch event {
+                    case .commandReceived(let deviceCommand):
+                        switch deviceCommand {
+                        case .tabReceived(_, let tabData):
+                            let title = tabData.entries.last?.title ?? ""
+                            let url = tabData.entries.last?.url ?? ""
+                            deferred.fill(Maybe(success: PushMessage.commandReceived(tab: ["title": title, "url": url])))
```

Comment:
Since we're refactoring this area, could we take the opportunity to remove deferred usage here?

Reply:
Yes! Filed a follow up in https://github.com/mozilla-mobile/firefox-ios/issues/14518, I'll get started on that once I'm done with one